src/librustdoc/html/render/mod.rs

```diff
@@ -2287,25 +2340,19 @@ fn sidebar_deref_methods(
     }
 }
 
-fn sidebar_struct(cx: &Context<'_>, buf: &mut Buffer, it: &clean::Item, s: &clean::Struct) {
-    let mut sidebar = Buffer::new();
-    let fields = get_struct_fields_name(&s.fields);
-
-    if !fields.is_empty() {
-        match s.ctor_kind {
-            None => {
-                print_sidebar_block(&mut sidebar, "fields", "Fields", fields.iter());
-            }
-            Some(CtorKind::Fn) => print_sidebar_title(&mut sidebar, "fields", "Tuple Fields"),
-            Some(CtorKind::Const) => {}
-        }
-    }
+#[derive(Template)]
+#[template(path = "sidebar/struct.html")]
+struct SidebarStruct {
+    fields: Vec<String>,
+    ctor_kind: Option<CtorKind>,
```

Comment:
The `CtorKind` is an internal implementation detail; what we want to express in our template struct is what information we plan to render. So this should be:

```suggestion
    fields_title: &'static str,
```

Containing one of "Fields", "Tuple Fields", or "".